File: src/main/java/cn/huangsy/interview64/SumNums.java
package cn.huangsy.interview64;

/**
 * @author huangsy
 * @date 2020/6/2 8:45
 */
public class SumNums {

    public int sumNums(int n) {
        int sum = n;
        boolean flag = n>0 && (sum+=sumNums(n-1)) >0;
        return sum;
    }
}

File: src/main/java/cn/leetcode/interview64/SumNums.kt
package cn.leetcode.interview64

/**
 * @author huangsy
 * @date 2020/6/2 8:43
 */
class SumNums {

    fun sumNums(n: Int): Int {
        var sum = n
        val flag = n > 0 && sumNums(n - 1).let { sum += it; sum } > 0
        return sum
    }
}